src/manage_nsfs/manage_nsfs_glacier.js

```diff
@@ -36,13 +36,38 @@ async function process_migrations() {
  */
 async function run_glacier_migrations(fs_context, backend) {
     // This WAL is getting opened only so that we can process all the prcess WAL entries
-    const wal = new PersistentLogger(
+    const wal = new PersistentLogger(config.NSFS_GLACIER_LOGS_DIR, GlacierBackend.MIGRATE_WAL_NAME, { locking: 'EXCLUSIVE' });
+    const failure_wal = new PersistentLogger(
         config.NSFS_GLACIER_LOGS_DIR,
-        GlacierBackend.MIGRATE_WAL_NAME,
-        { disable_rotate: true, locking: 'EXCLUSIVE' },
+        `${GlacierBackend.MIGRATE_WAL_NAME}.failure`,
+        { locking: 'EXCLUSIVE' },
     );
 
-    await wal.process_inactive(async file => backend.migrate(fs_context, file));
+    try {
+        // Process all the inactive and currently active log
+        await wal.process_inactive(async file => backend.migrate(fs_context, file, failure_wal.append.bind(failure_wal)));
+    } catch (error) {
+        console.error('failed to process restore logs:', error);
+    } finally {
+        await wal.close();
+    }
+
+    try {
+        // Process the inactive failure logs (don't process the current though)
+        // This will REMOVE the previous failure logs and will merge them with the current failures
+        await failure_wal.process_inactive(async file => backend.migrate(fs_context, file, failure_wal.append.bind(failure_wal)), false);
+    } catch (error) {
+        console.error('failed to process failure logs:', error);
+    }
+
+    try {
+        // Finally replace the current active so as to consume them in the next iteration
+        await failure_wal._replace_active();
+    } catch (error) {
+        console.error('failed to replace active failure log:', error);
+    }
+
+    await failure_wal.close();
```

Comment:
This entire flow seems to be identical to the run_glacier_restore with just few changed variables/callbacks.
Do you think we can write it as a one function that gets called with different args?